crawler.py

```diff
@@ -1,70 +1,93 @@
+    parsed_url = urlparse(url)
+    Domain = collections.namedtuple("Domain", "full_address, name")
+    return Domain(f"{parsed_url.scheme}://{parsed_url.netloc}", parsed_url.netloc)
+
+
+def page_title(url):
+    """Returns title of given url."""
+    browser = mechanize.Browser()
+    browser.set_handle_robots(False)
+    browser.open(url)
+    return browser.title()
+
+
+def page_links(url):
+    """Returns all links from given page."""
     browser = mechanize.Browser()
     browser.set_handle_robots(False)
-    browser.open(page_url)
+    browser.open(url)
+    domain = domain_tuple(url)
     links = set()
     for link in browser.links():
-        link = validated_link(link.url, domain)
-        # Check if link is not link to external site
-        if netloc in link:
-            links.add(link)
-    return [links, browser.title()]
-
-
-def map_page(page_url, domain, netloc):
-    '''Returns dictionary of page properties and page url.'''
-    links, title = page_properties(page_url, domain, netloc)
-
-    return {
-            page_url: {
-                'title': title,
-                'links': links}
-            }
-
-
-def map_site(url):
-    '''Returns dictionary of all pages properties from specific domain.'''
-    if validated(url):
-        parsed_url = urlparse(url)
-        Domain = namedtuple('Domain', 'domain, netloc')
-        url = Domain(parsed_url.scheme + '://' + parsed_url.netloc,
-                         parsed_url.netloc)
-        urls = [url.domain, ]  # comma added for more clarity
+        link = make_url_absolute(link.url, domain.full_address)
+        links.add(link)
+    return links
+
+
+def filtered_out_links(links, domain_name):
+    """Returns set of links that are only links from given domain."""
+    filtered_links = set()
+    # Check if link is not link to external site
+    for link in links:
+        if domain_name in link:
+            filtered_links.add(link)
+    return filtered_links
+
+
+def updated_urls_list(urls, links):
+    """Upade urls list with new links."""
+    for link in links:
+        if link not in urls:
+            urls.append(link)
+    return urls
+
+
+def site_map(url):
+    """Returns dictionary which will return page title and links of page from given domain."""
+    if is_validated(url):
+        domain = domain_tuple(url)
+        urls = [domain.full_address]
         site_mapping = dict()
+
         # Loop through pages from our domain
-        for url_ in urls:
-            page_mapping = map_page(url_, url.domain, url.netloc)
-            site_mapping.update(page_mapping)
-            for link in page_mapping[url_]['links']:
-                if link not in urls: urls.append(link)
+        for page_url in urls:
+            all_links = page_links(page_url)
+            links = filtered_out_links(all_links, domain.name)
+            title = page_title(page_url)
+            site_mapping.update({page_url: {"title": title, "links": links}})
```

Comment:
I wasn't sure if this was good idea to delete page_properties function, but it gave me apportunity to handle links of specific page and loop thorugh links to update urls list. Instead of this function I made two functions which return title and page links.

I'm also not sure about idea to split getting links into two functions, because getting all links(even this which aren't from given domain) is not useful in my case.
But on the other hand this gives possibility to modificate crawler.py to obtain other results like all links from pages, not only links from given url's domain.

Reply:
I think it was a good move to split `page_properties` into two functions. They are easier to read and understand now. 

Regarding having two functions for extracting links is also a good idea. One that fetches everything is required, as you do want to have everything so you can filter it down. Having filter function is also useful as you may extract links to other domains this way.